lib/hammer_cli_foreman/commands.rb

```diff
@@ -66,7 +66,7 @@ def self.record_to_common_format(data)
   class Command < HammerCLI::Apipie::Command
 
     def self.connection_name(resource_class)
-      CONNECTION_NAME
+      HammerCLIForeman.get_connection_name
```

Comment:
Why do we have different  connection names for different auth methods? This name should refer to which API we are connecting and should be immutable for a resource.

Reply:
We observed that [HammerCLIForeman.foreman_api_connection](https://github.com/theforeman/hammer-cli-foreman/pull/405/files) is being called multiple times when we run any of the login commands.

```
  def self.foreman_api_connection
    HammerCLI.context[:api_connection].create(CONNECTION_NAME) do
      HammerCLIForeman::Api::Connection.new(HammerCLI::Settings, Logging.logger['API'], HammerCLI::I18n.locale)
    end
  end
```


If we look at this method, it yields a block only if a previous connection with name {CONNECTION_NAME} (foreman) has not been called already.
From [HammerCli::Connection.rb](https://github.com/theforeman/hammer-cli/blob/c0c23e2f319f4f71c8294663a980e7830e8754e7/lib/hammer_cli/connection.rb#L21):
```
def create(name, &create_connector_block)
      unless connections[name]
        connector = yield
        @logger.debug("Registered: #{name}") if @logger
        connections[name] = connector
      end
      connections[name]
end
```

This worked fine in our initial flow since there was only a single type of authentication flow available (ie basic auth).

When we introduced more authentication ways, and called `HammerCLIForeman.foreman_api_connection` from our [auth.rb](https://github.com/theforeman/hammer-cli-foreman/blob/347bf39cf73f50830c431da99e75bdc581f0a831/lib/hammer_cli_foreman/auth.rb#L9) file, for other authentication ways, the block was never yielded, because the first time `HammerCLIForeman.foreman_api_connection` was called, the connection (with basic_auth) was already stored with `FOREMAN` key. 

Hence we did these changes, so that the block is yielded, when we use any other authentiction way except basic_auth. 

I will be glad to explain this in more detail, along with the flow as required.